pkg/arvo/lib/easy-print.hoon

```diff
@@ -0,0 +1,481 @@
+    ?~  gim
+      ~
+    =+  myd=$(p.q.ham t.p.q.ham, lum +.lum)
+    ?~  myd
+      ~
+    [~ u.gim u.myd]
+  ::
+      {$pear *}
+    ?.  =(lum q.q.ham)
+      ~
+    =.  p.q.ham
+      (rash p.q.ham ;~(sfix (cook crip (star low)) (star hig)))
+    =+  fox=$(q.ham [%mato p.q.ham])
+    ?>  ?=({~ $leaf ^} fox)
+    ?:  ?=(?($n $tas) p.q.ham)
+      fox
+    [~ %leaf '%' p.u.fox]
+  ::
+      {$stop *}
+    ?:  (~(has in gil) [p.q.ham lum])  ~
+    =+  kep=(~(get by p.ham) p.q.ham)
+    ?~  kep
+      ~|([%stop-loss p.q.ham] !!)
+    $(gil (~(put in gil) [p.q.ham lum]), q.ham u.kep)
+  ::
+      {$tree *}
+    =-  ?~  tok
+          ~
+        [~ %rose [[' ' ~] ['{' ~] ['}' ~]] u.tok]
+    ^=  tok
+    =+  tuk=*(list tank)
+    |-  ^-  (unit (list tank))
+    ?:  =(~ lum)
+      [~ tuk]
+    ?.  ?=({n/* l/* r/*} lum)
+      ~
+    =+  rol=$(lum r.lum)
+    ?~  rol
+      ~
+    =+  tim=^$(q.ham q.q.ham, lum n.lum)
+    ?~  tim
+      ~
+    $(lum l.lum, tuk [u.tim u.rol])
+  ::
+      {$unit *}
+    ?@  lum
+      ?.(=(~ lum) ~ [~ %leaf '~' ~])
+    ?.  =(~ -.lum)
+      ~
+    =+  wal=$(q.ham q.q.ham, lum +.lum)
+    ?~  wal
+      ~
+    [~ %rose [[' ' ~] ['[' ~] [']' ~]] [%leaf '~' ~] u.wal ~]
+  ==
+::
+++  doge
+  |=  ham/cape
+  =-  ?+  woz  woz
+        {$list * {$mato $'ta'}}  %path
+        {$list * {$mato $'t'}}   %wall
+        {$list * {$mato $'tD'}}  %yarn
+        {$list * $yarn}          %wool
+      ==
+  ^=  woz
+  ^-  wine
+  ?.  ?=({$stop *} q.ham)
+    ?:  ?&  ?=  {$bswt {$pear $n $0} {$plot {$pear $n $0} {$face *} ~} ~}
+              q.ham
+            =(1 (met 3 p.i.t.p.i.t.p.q.ham))
+        ==
+      [%unit =<([p q] i.t.p.i.t.p.q.ham)]
+    q.ham
+  =+  may=(~(get by p.ham) p.q.ham)
+  ?~  may
+    q.ham
+  =+  nul=[%pear %n 0]
+  ?.  ?&  ?=({$bswt *} u.may)
+          ?=({* * ~} p.u.may)
+          |(=(nul i.p.u.may) =(nul i.t.p.u.may))
+      ==
+    q.ham
+  =+  din=?:(=(nul i.p.u.may) i.t.p.u.may i.p.u.may)
+  ?:  ?&  ?=({$plot {$face *} {$face * $stop *} ~} din)
+          =(p.q.ham p.q.i.t.p.din)
+          =(1 (met 3 p.i.p.din))
+          =(1 (met 3 p.i.t.p.din))
+      ==
+    :+  %list
+      (cat 3 p.i.p.din p.i.t.p.din)
+    q.i.p.din
+  ?:  ?&  ?=  $:  $plot
+                  {$face *}
+                  {$face * $stop *}
+                  {{$face * $stop *} ~}
+              ==
+              din
+          =(p.q.ham p.q.i.t.p.din)
+          =(p.q.ham p.q.i.t.t.p.din)
+          =(1 (met 3 p.i.p.din))
+          =(1 (met 3 p.i.t.p.din))
+          =(1 (met 3 p.i.t.t.p.din))
+      ==
+    :+  %tree
+      %^    cat
+          3
+        p.i.p.din
+      (cat 3 p.i.t.p.din p.i.t.t.p.din)
+    q.i.p.din
+  q.ham
+::
+++  dole
+  ^-  cape
+  =+  gil=*(set type)
+  =+  dex=[p=*(map type @) q=*(map @ wine)]
+  =<  [q.p q]
+  |-  ^-  {p/{p/(map type @) q/(map @ wine)} q/wine}
+  =-  [p.tez (doge q.p.tez q.tez)]
+  ^=  tez
+  ^-  {p/{p/(map type @) q/(map @ wine)} q/wine}
+  ?:  (~(meet ut sut) -:!>(*type))
```

Comment:
I just tried this branch out locally. Hitting tab on an empty buffer (or immediately before `:title`, for instance) results in a crash on this line.

Reply:
Whoops, that was working before.  I'm guessing this broke because now we render everything to tanks instead of only if fewer than 10; there's probably some type that breaks this.  Will investigate.

Reply:
Hmm, any idea why putting the meet call in +mule take it from "super fast" to "painfully slow"?

Reply:
FWIW, it's crashing for the same reason that `*pfix` crashes (don't know why that crashes).

Reply:
I would guess the +mule turns a cache hit into a miss.

Reply:
It's actually quite strange to me that +pfix compiles at all, since it takes a noun and then passes that into +comp, which is expecting something that has a `q.vex`.  This seems like a bug in `+mint:et`, but I'm not sure I want to track that down right now.

I changed `+pfix` to have an appropriate sample type and the crash went away, so I'll check if anything else has the same problem.  If there's just a couple that I can fix, I'll do that and leave off the +mule.

Reply:
+sfix is defined in the same way. This seems related to #1714

Reply:
Github's UI is fritzing out on me, not sure if this message went through:

It's actually quite strange to me that +pfix compiles at all, since it takes a noun and then passes that into +comp, which is expecting something that has a q.vex. This seems like a bug in +mint:et, but I'm not sure I want to track that down right now.

I changed +pfix to have an appropriate sample type and the crash went away, so I'll check if anything else has the same problem. If there's just a couple that I can fix, I'll do that and leave off the +mule.

Reply:
I can fix +sfix, I haven't figured out +perk yet.  I checked all the initial letters and they work except `p` and `s`.

Reply:
The issue with +perk is strange: `$:perk` fails (mull), while `(perk ~)` succeeds. But `$:$:=>(perk .(+6 ~))` succeeds, while `$:(perk ~)` fails (nest).

Reply:
Going to keep looking at why `*perk` doesn't compile, but it occurs to me that maybe wet gates shouldn't show their output type, since it may change depending on the input.  In any case, wet gates should probably be written differently (maybe replace `->` with `~>`).

Reply:
Eh... fixed!

https://github.com/urbit/urbit/blob/0014d1cf2b9a7b34a346801d91057fa50eccbab2/pkg/arvo/app/dojo.hoon#L1147-L1148

This feels like a compiler bug and I've spent a few hours trying to track it down without any success, so I'm going to declare it out of scope.